neet/statespace.py

```diff
@@ -191,81 +194,77 @@ def decode(self, x):
             for i in range(self.ndim):
-                state[i] = x % self.bases[i]
-                x = int(x / self.bases[i])
+                base = self.bases[i]
+                state[i] = encoded % base
+                encoded = int(encoded / base)
         return state
 
-    def check_states(self, states):
+    def __contains__(self, states):
         """
-        Check the validity of the provided states.
+        Determine if a state is in the state space
 
-        .. rubric:: Examples;
+        .. rubric:: Examples:
 
         ::
 
-            >>> StateSpace(3).check_states([0,0,0])
+            >>> state_space = StateSpace(3)
+            >>> [0, 0, 0] in state_space
             True
-            >>> StateSpace(3).check_states([0,0])
-            Traceback (most recent call last):
-                ...
-            ValueError: incorrect number of states in array
-            >>> StateSpace(3).check_states([1,2,1])
-            Traceback (most recent call last):
-                ...
-            ValueError: invalid node state in states
-
-            >>> StateSpace([2, 3, 2]).check_states([0, 2, 1])
+            >>> [0, 0] in state_space
+            False
+            >>> [1, 2, 1] in state_space
+            False
+
+            >>> [0, 2, 1] in StateSpace([2, 3, 2])
             True
-            >>> StateSpace([2, 2, 3]).check_states([0, 1])
-            Traceback (most recent call last):
-                ...
-            ValueError: incorrect number of states in array
-            >>> StateSpace([2, 3, 4]).check_states([1, 1, 6])
-            Traceback (most recent call last):
-                ...
-            ValueError: invalid node state in states
-
-        :returns: ``True`` if the ``states`` are valid, otherwise an error is raised
+            >>> [0, 1] in StateSpace([2, 2, 3])
+            False
+            >>> [1, 1, 6] in StateSpace([2, 3, 4])
+            False
+
         :param states: the one-dimensional sequence of node states
-        :type states: sequence
-        :raises TypeError: if ``states`` is not iterable
-        :raises ValueError: if ``len(states)`` is not the number of nodes in the network
-        :raises ValueError: if ``states[i] not in [0,1]`` for any node ``i``
+        :returns: ``True`` if the ``states`` are valid
         """
-        if len(states) != self.ndim:
-            raise ValueError("incorrect number of states in array")
-
-        if self.is_uniform:
-            for x in states:
-                if x not in range(self.base):
-                    raise ValueError("invalid node state in states")
-        else:
-            for x, b in zip(states, self.bases):
-                if x not in range(b):
-                    raise ValueError("invalid node state in states")
+        try:
+            if len(states) != self.ndim:
+                return False
+
+            if self.is_uniform:
+                for state in states:
+                    if self.base <= state or state < 0:
```

Comment:
I would change line 259 to
`if base not in range(self.base):` or non-integers could pass the check.